cumulusci/cli/cci.py

```diff
@@ -259,42 +253,93 @@ def version():
     click.echo()
 
 
-@main.command(name="shell", help="Drop into a Python shell")
-def shell():
-    try:
-        config = load_config(load_project_config=True, load_keychain=True)
-    except SystemExit:
-        config = load_config(load_project_config=False, load_keychain=False)
+@cli.command(name="shell", help="Drop into a Python shell")
+@pass_runtime(require_project=False)
+def shell(runtime):
+    # alias for backwards-compatibility
+    config = runtime
 
     code.interact(local=dict(globals(), **locals()))
 
 
+GIST_404_ERR_MSG = """A 404 error code was returned when trying to create your gist.
+Please ensure that your GitHub personal access token has the 'Create gists' scope."""
+CCI_LOG_NOT_FOUND_MSG = """No logfile to open at path: {}
+Please ensure you're running this command from the same directory you were experiencing an issue."""
+LAST_CMD_HEADER = "\n\n\nLast Command Run\n================================\n"
+CCI_LOGFILE_PATH = Path.home() / ".cumulusci/logs/cci.log"
+
+
+@cli.command(name="gist", help="Create a GitHub gist from the latest logfile")
+@pass_runtime(require_project=False)
+def gist(runtime):
+    try:
+        last_cmd_log = open(CCI_LOGFILE_PATH, "r")
+    except FileNotFoundError:
+        error_msg = CCI_LOG_NOT_FOUND_MSG.format(CCI_LOGFILE_PATH)
+        click.echo(error_msg)
+        raise CumulusCIException(error_msg)
+
+    log_content = last_cmd_log.read()
+    filename = f"cci_output_{datetime.utcnow()}.txt"
+    files = {
+        filename: {"content": f"{get_context_info()}{LAST_CMD_HEADER}{log_content}"}
+    }
+
+    try:
+        gh = RUNTIME.keychain.get_service("github")
+        gist = create_gist(
+            get_github_api(gh.config["username"], gh.config["password"]),
+            "CumulusCI Error Output",
+            files,
+        )
+    except Exception as e:
+        if e.response.status_code == 404:
+            click.echo(GIST_404_ERR_MSG)
+        else:
+            click.echo(f"An error occurred attempting to create your gist:\n{e}")
+            sys.exit(1)
```

Comment:
For both lines 298 and 300 I think it would be more consistent to raise an exception and let the wrapper take care of printing it and exiting?